src/test/java/com/mongodb/jdbc/MongoSQLResultSetTest.java

```diff
@@ -262,17 +277,40 @@ void testGetStringValues() throws Exception {
         // BINARY_COL              [10, 20, 30]
         // ANY_OF_INT_STRING_COL   3
         // INT_OR_NULL_COL         null
+        // NULL_COL                null
         // INT_COL                 4
         // ANY_COL                 "{}"
         // ARRAY_COL               [5, 6, 7]
-        //
+        // DOC_COL                 {"c": 5}
 
         //Test String values are as expected
         assertEquals("2.4", mongoSQLResultSet.getString(DOUBLE_COL_LABEL));
         assertEquals("b", mongoSQLResultSet.getString(STRING_COL_LABEL));
+        assertNull(mongoSQLResultSet.getString(INT_NULLABLE_COL_LABEL));
         assertNull(mongoSQLResultSet.getString(NULL_COL));
         assertEquals("4", mongoSQLResultSet.getString(INT_COL_LABEL));
-        assertEquals(null, mongoSQLResultSet.getString(ANY_COL_LABEL));
+        assertNull(mongoSQLResultSet.getString(ANY_COL_LABEL));
+        assertEquals("[5, 6, 7]", mongoSQLResultSet.getString(ARRAY_COL_LABEL));
+        assertEquals("{\"c\": 5}", mongoSQLResultSet.getString(DOC_COL_LABEL));
+
+        // Check that getObject().toString() matches getString()
```

Comment:
This is the primary place where I added new testing. By adding the document value `doc`, I am covering the two cases this currently primarily impacts: arrays and documents.

Reply:
We should aim for covering all BSON types and on top of checking that getString and getObjectToString match, we should also make sure that the values are valid relaxed extended JSON. This can be done with something like `Document.parse("{\"val\":" + extendedJsonVal + "}")`. If the parsing does not fail, then it means extendedJsonVal is a valid extended json value.